data/courses/addCourseScript.php

```diff
@@ -8,49 +9,78 @@
     // Use extract method to set variables by default from post
     extract($_POST);
 
-    // Format days appropriately for DB
-    $number_of_days = count($_POST['days']);
-    $days = "";
-    $daysArray = $_POST['days'];
-    for ($i = 0; $i < $number_of_days; $i++) {
-        $days = $days . $daysArray[$i];
-    }
+    $course_code = test_input($_POST['course_code']);
+    $id = test_input($_POST['id']);
+    $semester = test_input($_POST['semester']);
 
     $conn = new mysqli(DB_SERVER, DB_USER, DB_PASS, DB_NAME);
 
     if ($conn->connect_error) {
         die("Connection failed: " . $conn->connect_error);
     }
 
-    if ($conn) {
-        $SQL = $conn->prepare("INSERT INTO course (course_code,title,semester,days,time,instructor,room,start_date,end_date) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)");
+    if($conn)
+    {
+        // TODO: Store id in session & remove this section (optional later)
+        $SQL = $conn->prepare("SELECT id FROM `student` WHERE id=?"); //Ensure student actually exists
+        $SQL->bind_param('i', $id);
+        $SQL->execute();
+        $SQL->store_result();
+        $result = $SQL->num_rows;
+        $SQL->close();
+        if($result <= 0) { //Student does not exist
+            header("Location: ".PATH_ADD_COURSE."?stat=addCourseES");
+            exit();
+        }
+
+        $SQL = $conn->prepare("SELECT course_code, semester FROM `course` WHERE course_code=? AND semester=?"); //Ensure course actually exists
+        $SQL->bind_param('ss', $course_code, $semester);
+        $SQL->execute();
+        $SQL->store_result();
+        $result = $SQL->num_rows;
+        $SQL->close();
+        if($result <= 0) { //Course does not exist
+            header("Location: ".PATH_ADD_COURSE."?stat=addCourseEC");
+            exit();
+        }
+
+        $SQL = $conn->prepare("SELECT course_code, id, semester FROM `registrar` WHERE course_code=? AND id=? AND semester=?"); //Ensure registrar doesn't actually exists
+        $SQL->bind_param('sis', $course_code, $id, $semester);
+        $SQL->execute();
+        $SQL->store_result();
+        $result = $SQL->num_rows;
+        $SQL->close();
+        if($result >= 1) { //Registrar already exists
+            header("Location: ".PATH_ADD_COURSE."?stat=addCourseER");
+            exit();
+        }
 
+        $SQL = $conn->prepare("SELECT id FROM `registrar` WHERE id=?"); //Ensure max number of registered courses not reached
+        $SQL->bind_param('i', $id);
+        $SQL->execute();
+        $SQL->store_result();
+        $result = $SQL->num_rows;
+        $SQL->close();
+
+        // Business layer to check if max number of courses reached
+        require_once('../../business/courses/addCourseBusiness.php');
```

Comment:
File doesn't exist.  Getting an error because of it when I try adding a course for a student.

Reply:
Somehow it didn't push the new file I made, pushing fix soon.

Reply:
The commit didn't add the new file, pushing soon.